source/entrance.cpp

```diff
@@ -855,10 +1205,18 @@ void CreateEntranceOverrides() {
     PlacementLog_Msg(message);
 
     s16 originalIndex = entrance->GetIndex();
-    s16 destinationIndex = entrance->GetReverse()->GetIndex();
     s16 originalBlueWarp = entrance->GetBlueWarp();
     s16 replacementIndex = entrance->GetReplacement()->GetIndex();
-    s16 replacementDestinationIndex = entrance->GetReplacement()->GetReverse()->GetIndex();
+
+    s16 destinationIndex = -1;
+    s16 replacementDestinationIndex = -1;
+
+    // Only set destination indices for two way entrances and when decouple entrances
+    // is off
+    if (entrance->GetReverse() != nullptr && !Settings::DecoupleEntrances) {
+      replacementDestinationIndex = entrance->GetReplacement()->GetReverse()->GetIndex();
+      destinationIndex = entrance->GetReverse()->GetIndex();
+    }
```

Comment:
I agree with changing `Src` and `Dest` to `From` and `To`.

`SaveFile_SetEntranceDiscovered` doesn't need to be modified because decoupled entrances have their `destination` and `overrideDestination` fields set to -1. So when the function gets called a second time for the `overrideDestination` it just returns without changing anything.